aptos-move/block-executor/src/executor.rs

```diff
@@ -506,13 +529,43 @@ where
         patched_resource_write_set
     }
 
+    fn map_id_to_values_in_read_set_sequential(
+        delayed_field_keys: Option<impl Iterator<Item = T::Identifier>>,
+        write_set_keys: HashSet<T::Key>,
+        read_set: RefCell<HashMap<T::Key, HashSet<T::Identifier>>>,
+        unsync_map: &UnsyncMap<T::Key, T::Value, X, T::Identifier>,
+        latest_view: &LatestView<T, S, X>,
+    ) -> HashMap<T::Key, T::Value> {
+        let mut patched_resource_write_set = HashMap::new();
+        if let Some(delayed_field_keys) = delayed_field_keys {
+            let delayed_field_keys = delayed_field_keys.collect::<HashSet<_>>();
+            for (key, delayed_field_keys_in_resource) in read_set.borrow().iter() {
+                if write_set_keys.contains(key)
+                    || delayed_field_keys.is_disjoint(delayed_field_keys_in_resource)
```

Comment:
Don't we have the check for `is_disjoint` that in `replace_ids_with_values` already? Also seems like `delayed_field_keys_in_resource` are computed twice, before and here in `replace_ids_with_values`?

Reply:
Good catch. Done.